### PR TITLE
Change maxBufferLength to pause rather than drop writes

### LINKED ISSUE #83: Change maxBufferLength logic
https://github.com/Level/level-ws/blob/182314539e46f1b0c056a622c676612ad005b447/level-ws.js#L32-L35

Rather than ignoring writes, we can pause until the current buffer is flushed. Along the lines of:

```js
self.once('_flush', function (err) {
  if (err) return next(err)
  self._write(d, enc, next)
}) 
```

### PR DESCRIPTION
**Pull request by vweevers**
Closes #83.

The approach I suggested in https://github.com/Level/level-ws/issues/83#issuecomment-401539523 (pushing to the buffer on every `_write` regardless of `maxBufferLength `) didn't work out because if a flush was already scheduled, pushing to the buffer will make it spill (doing batches that are `maxBufferLength + x` in length).

### CODE REVIEW

**Comment by vweevers** [comment]
`Could not resolve host: nodejs.org` error on Travis, restarting.

File: level-ws.js [hunk 1]
```diff
@@ -9,12 +9,18 @@ function WriteStream (db, options) {
...
+  Writable.call(this, {
+    objectMode: true,
+    highWaterMark: options.highWaterMark || 16
+  })
+
+  this._options = options
   this._db = db
   this._buffer = []
   this._flushing = false
+  this._maxBufferLength = options.maxBufferLength || Infinity
```
Comment (ralphtheninja) [inline level-ws.js]: Nice one

File: test.js [hunk 2]
```diff
@@ -422,3 +422,54 @@ test('test that missing type errors', function (t, ctx, done) {
...
+})
+
+function testMaxBuffer (max, randomize) {
+  return function (t, ctx, done) {
+    var ws = WriteStream(ctx.db, { maxBufferLength: max })
+    var sourceData = []
+    var batches = []
+
+    for (var i = 0; i < 20; i++) {
+      sourceData.push({ key: i < 10 ? '0' + i : String(i), value: 'value' })
```
Comment (ralphtheninja) [inline test.js]: I guess this to make i < 10 sort below "10" right?

Comment (vweevers) [reply test.js]: Yes

**Pull request merged**